collab-database/src/workspace_database/body.rs

```diff
@@ -268,6 +256,32 @@ impl WorkspaceDatabaseBody {
     })
   }
 
+  pub fn add_database(&self, txn: &mut TransactionMut, database_id: &str, view_ids: Vec<String>) {
+    let linked_views: HashSet<String> = view_ids.into_iter().collect();
+    let record = DatabaseMeta {
+      database_id: database_id.to_string(),
+      created_at: timestamp(),
+      linked_views: linked_views.into_iter().collect(),
+    };
+    self.push_back(txn, record);
+  }
+
+  pub fn batch_add_database(
+    &mut self,
+    txn: &mut TransactionMut,
+    view_ids_by_database_id: HashMap<String, Vec<String>>,
+  ) {
+    for (database_id, view_ids) in view_ids_by_database_id {
+      let linked_views: HashSet<String> = view_ids.into_iter().collect();
+      let record = DatabaseMeta {
+        database_id,
+        created_at: timestamp(),
+        linked_views: linked_views.into_iter().collect(),
+      };
+      self.push_back(txn, record);
```

Comment:
I don't know if appflowy-collab exposes this method, but in general for batch inserts of scalar types, yrs offers [insert_range](https://docs.rs/yrs/latest/yrs/types/array/trait.Array.html#method.insert_range) method which more optimised for inserting sequences.